### ISSUE TITLE
Cast all integer keys to strings for consistency

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6285]
We have the same `@todo` item in our `form_process_checkboxes()` as in https://www.drupal.org/project/drupal/issues/3293550, so I'm filing this to not forget about it (and to see if backporting the code from what gets in Drupal fixes it for us as well - might need to wait till Backdrop 2.0 though).

**Comment by kiamlaluno** [comment]
Truly, that `@todo` does not even make sense in Drupal 7.

> @todo For Drupal 8, cast all integer keys to strings for consistency with `form_process_radios()`.

`form_process_radios()` does not cast integer keys to strings, neither in [Drupal 7](https://api.drupal.org/api/drupal/includes%21form.inc/function/form_process_radios/7.x) nor [Backdrop](https://docs.backdropcms.org/api/backdrop/core%21includes%21form.inc/function/form_process_radios/1).

**Comment by kiamlaluno** [comment]
Basing on a comment in [`form_process_radios()`](https://docs.backdropcms.org/api/backdrop/core%21includes%21form.inc/function/form_process_radios/1) the key value is sanitized in [`backdrop_attributes()`](https://docs.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/backdrop_attributes/1), but that does not seem to mean `$key` is casted to a string everywhere.

> The key is sanitized in backdrop_attributes() during output from the theme function.

**Comment by klonos** [comment]
Right. So maybe we should remove that `@todo` then 🤔 ...however, I came across this issue while doing some research around why values of various form elements are being saved in config as strings, despite having specified the keys in their `'#options'` as numeric. See #2425.
